src/Symfony/Component/Serializer/Exception/MissingConstructorArgumentsException.php

```diff
@@ -17,19 +17,52 @@
 class MissingConstructorArgumentsException extends RuntimeException
 {
     /**
+     * @deprecated since Symfony 6.3
+     *
      * @var string[]
      */
     private $missingArguments;
 
-    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [])
+    private ?string $class;
+    private ?string $missingArgument;
+
+    public function __construct(string $message, int $code = 0, \Throwable $previous = null, /* string */ $class = null, string $missingArgument = null)
```

Comment:
I personally don't think that we might reduce multiple arguments to a single argument.
IMHO, this class is named `MissingConstructorArgumentsException` and we could reasonably think that users are already using it in their user land with multiple arguments.

Reply:
I think that's part of the problem, the naming is very misleading. I was expecting the `getMissingConstructorArguments()` method to return all missing arguments, but after fixing the only one in the array, a new one showed up. Maybe the entire exception should be renamed as its name doesn't match how it's used. As for user land code, users can always extend the exception and add multiple arguments, or just create their own exception.